src/shared/install-printf.c

```diff
@@ -57,8 +58,8 @@ static int specifier_instance(char specifier, void *data, void *userdata, char *
         if (r < 0)
                 return r;
 
-        if (!instance) {
-                instance = strdup("");
+        if (isempty(instance) && i->default_instance) {
+                instance = strdup(i->default_instance);
                 if (!instance)
                         return -ENOMEM;
         }
```

Comment:
hmm, with this change, if unit_name_to_instance() returns NULL in the ret parameter it will be passed back as NULL, and I doubt we want that. And yes, unit_name_to_instance() does return NULL in the ret parameter if the unit name does not contain an `@`...